File: LSP/Delphi/aplicado/Teste.pas
unit Teste;

interface

uses
  TestFramework, Unit1;

type
  TestTRetangulo = class(TTestCase)
  published
    procedure AreaDeveSer6QuandoRetangulo_2X3;
    procedure AreaDeveSer9QuandoQuadrado_3X3;
    procedure AreaDeveSer20QuandoQuadradoRetangulo_4X5;
  end;

implementation

procedure TestTRetangulo.AreaDeveSer6QuandoRetangulo_2X3;
var
  retangulo: IForma;
  retangulo: TRetangulo;
begin
  retangulo := TRetangulo.Create;
  retangulo.Altura := 2;
  retangulo.Largura := 3;
  forma := retangulo;
  CheckEquals(6, forma.Area);
end;

procedure TestTRetangulo.AreaDeveSer9QuandoQuadrado_3X3;
var
  forma: IForma;
  quadrado: TQuadrado;
begin
  quadrado := TQuadrado.Create;
  quadrado.TamanhoLado := 3;
  forma := quadrado;
  CheckEquals(9, forma.Area);
end;

procedure TestTRetangulo.AreaDeveSer20QuandoQuadradoRetangulo_4X5;
var
  forma: IForma;
  quadrado: TQuadrado;
begin
  quadrado := TQuadrado.Create;
  quadrado.TamanhoLado := 4; // não faz mais sentido
  quadrado.TamanhoLado := 5; // não faz mais sentido
  forma := quadrado;
  CheckEquals(20, forma.Area); // não faz mais sentido
end;

initialization

RegisterTest(TestTRetangulo.Suite);

end.

File: LSP/Ruby/aplicado/teste.rb
require 'test/unit'

class Teste < Test::Unit::TestCase
    def area_deve_ser_6_quando_retangulo_2x3
    	retangulo = Retangulo.new
		retangulo.altura = 2
		retangulo.largura = 3
		assert_equal 6, retangulo.area
    end
    
	def area_deve_ser_9_quando_quadrado_3x3
		quadrado = Quadrado.new
		quadrado.tamanhoLado = 3
		assert_equal 9, quadrado.area
    end
	
	def area_deve_ser_20_quando_quadrado_retangulo_4x5
		quadrado = Quadrado.new
		quadrado.tamanhoLado = 4 # não faz mais sentido
		quadrado.tamanhoLado = 5 # não faz mais sentido
		assert_equal 20, quadrado.area # não faz mais sentido
    end
end